src/Rings/mpoly-ideals.jl

```diff
@@ -1191,17 +1197,27 @@ function minimal_primes(
 
   # This will in many cases lead to an easy simplification of the problem
   if factor_generators
-    J = typeof(I)[ideal(R, elem_type(R)[])]
+    J = [ideal(R, gens(I))] # A copy of I as initialization
     for g in gens(I)
       K = typeof(I)[]
       is_zero(g) && continue
       for (b, k) in factor(g)
+        # Split the already collected components with b
         for j in J
           push!(K, j + ideal(R, b))
         end
       end
       J = K
     end
+
+    unique_comp = typeof(I)[]
+    for q in J
+      is_one(q) && continue
+      q in unique_comp && continue
+      push!(unique_comp, q)
+    end
+    J = unique_comp
```

Comment:
@HechtiDerLachs I resolved the merge conflicts -- I hope I didn't screw that up too badly...